### ISSUE TITLE
Avoiding the potential for conflict between map and modify

### ISSUE THREAD

**Issue by brettz9** [aaronpowell/db.js#153]
I'm seeking to understand--and document--the relative priority of `map()` and `modify()` (since `map` can optionally precede `modify`).

There is this code block inside of `runQuery`:

```js
                        if (matchFilter) {
                            counter++;
                            results.push(mapper(result));
                            // if we're doing a modify, run it now
                            if (modifyObj) {
                                result = modifyRecord(result);
                                cursor.update(result);
                            }
                        }
                        cursor.continue();
```

...with `results` being subsequently returned with a successful promise resolution (on the completion of the transaction).

To determine the effects of one on the other, we don't need to worry about `modifyRecord` (unless it is ever changed to not work fully by reference).

However, if `mapper` (set by `.map()`) does not act by reference (such as in the README example), the results returned to the promise will differ from the actual update.

While this appears to be by design (since `mapper`'s intent is apparently in how the object is returned rather than stored), I think this distinction should be clarified.

Also, there is the potential for confusion if the mapper ever acts by reference on the passed object, since it *will* actually modify the data to be stored in such a case. Although a note could be added to clarify this, I think it would be safer to clone the result object before passing it to the mapper to thereby ensure that a mapper cannot accidentally modify the object stored (while modify continues not to effect what is returned):

```js
                            results.push(mapper(Object.assign({}, result)));
```

**Comment by brettz9** [comment]
In my PR, I realized we could not clone with `Object.assign`, so I handled it differently (and in a clearer way anyhow).

**Comment by brettz9** [comment]
Addressed by PR #154 , closing, thanks!
